### PR TITLE
pure-maps: 3.3.0 -> 3.4.0

### PR DESCRIPTION
**Pull request by dotlambda**
Diff: https://github.com/rinigus/pure-maps/compare/3.3.0...3.4.0

Changelog: https://github.com/rinigus/pure-maps/blob/3.4.0/NEWS.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351558`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>kdePackages.maplibre-native-qt (qt6Packages.maplibre-native-qt)</li>
    <li>libsForQt5.mapbox-gl-qml (plasma5Packages.mapbox-gl-qml)</li>
    <li>libsForQt5.maplibre-native-qt (plasma5Packages.maplibre-native-qt)</li>
    <li>pure-maps</li>
    <li>s2geometry</li>
  </ul>
</details>

**Comment by Scrumplex** [comment]
This segfaults for me, while the version on master works fine.

https://gist.github.com/Scrumplex/33dfe6dfcf8e9e7a5b2ceb35a540a822

**Comment by sikmir** [comment]
https://github.com/NixOS/nixpkgs/pull/351558/commits/c640471d9f000c7339448e3e7919cc2e1ba96d3a contains changes to mapbox-gl-qml, I guess it should be in another commit.

**Comment by dotlambda** [comment]
> This segfaults for me

can't reproduce

File: pkgs/development/libraries/mapbox-gl-qml/default.nix [hunk 1]
```diff
@@ -1,38 +1,48 @@
...
-, curl
-, qtbase
-, qtlocation
-, maplibre-gl-native
+{
+  cmake,
+  fetchFromGitHub,
+  lib,
+  maplibre-native-qt,
+  qtbase,
```
Comment (K900) [inline pkgs/development/libraries/mapbox-gl-qml/default.nix:6]: nit: qtbase seems unused?

**Review by K900 — COMMENTED** [review]
Qt bits LGTM, assuming the rest works.

**Review by Scrumplex — APPROVED** [review]
Not sure what caused the segfault before. Seems to work now

File: pkgs/development/libraries/mapbox-gl-qml/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/libraries/mapbox-gl-qml/default.nix:6]: It's used in `qtbase.qtQmlPrefix`.

**Pull request merged**